### PR TITLE
adding support for @ResponseStatus annotation into spring-http-interface java spring generator

### PR DESCRIPTION
**Pull request by alfet81**
Spring-http-interface generator is missing the support for useResponseEntity option for adding @ResponseStatus annotation similar to Java Spring generator. This feature was probably omitted due to the fact that the HTTP interface is mainly used for client side implementations. But there are cases when it also can be used for server side implementations like for example a REST controller which plays a role of a server side implementation thus other micro-services can use the same HTTP interface to make calls to the server-side endpoints utilizing same API. See https://docs.spring.io/spring-framework/reference/integration/rest-clients.html#rest-http-interface.

The change should be made in api.mustache template.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming `7.x.0` minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by alfet81** [comment]
@bbdouglas @sreeshas @jfiala @lukoyanov @cbornet  @jeff9finger @karismann ) @Zomzog  @lwlee2608 @martin-mfg could anybody review and approve. We need this feature ASAP. Thanks.

**Comment by alfet81** [comment]
Is it possible to push this change into 7.10 milestone so we can have the feature before the year 2025?

**Review by martin-mfg — APPROVED** [review]

**Comment by wing328** [comment]
thanks for the PR. can we include samples using this option and test it in the CI workflow?

a good starting is to copy `bin/configs/spring-http-interface-reactive.yaml`

```
$ cat ./bin/configs/spring-http-interface-reactive.yaml
generatorName: spring
library: spring-http-interface
outputDir: samples/client/petstore/spring-http-interface-reactive
inputSpec: modules/openapi-generator/src/test/resources/3_0/spring/petstore-with-fake-endpoints-models-for-testing.yaml
templateDir: modules/openapi-generator/src/main/resources/JavaSpring
additionalProperties:
  artifactId: spring-http-interface-reactive
  snapshotVersion: "true"
  hideGenerationTimestamp: "true"
  reactive: "true"
  # documentation provider should be ignored
  documentationProvider: "springfox"
  # annotation provider should be ignored
  annotationLibrary: "swagger1"
  # validation should be ignored
  useBeanValidation: "true"
  performBeanValidation: "true"

```

then add the new sample folder to the workflow: .github/workflows/samples-jdk17.yaml

**Comment by alfet81** [comment]
@wing328 Thanks for your comment. I've pushed couple extra commits per your request. Please let me know if they look fine. Thanks.

**Comment by wing328** [comment]
https://github.com/OpenAPITools/openapi-generator/actions/runs/11790584074/job/32930532262?pr=20071

can you please update the samples to fix the CI failure?

File: modules/openapi-generator/src/test/resources/3_0/spring/petstore-with-fake-endpoints-models-for-testing.yaml [hunk 1]
```diff
@@ -26,7 +26,7 @@ paths:
       requestBody:
         $ref: "#/components/requestBodies/Pet"
       responses:
-        "200":
+        "201":
```
Comment (martin-mfg) [inline modules/openapi-generator/src/test/resources/3_0/spring/petstore-with-fake-endpoints-models-for-testing.yaml:29]: Is there a reason to make any changes in this file? If not, could you revert them please?

**Review by martin-mfg — CHANGES_REQUESTED** [review]

File: .github/workflows/samples-jdk17.yaml [hunk 2]
```diff
@@ -44,15 +48,15 @@ jobs:
...
+          - samples/client/petstore/spring-http-interface-noResponseEntity
           - samples/client/petstore/java/webclient-jakarta
           - samples/client/petstore/java/microprofile-rest-client-outer-enum
           # servers
           - samples/openapi3/server/petstore/springboot-3
           - samples/server/petstore/java-camel/
           - samples/server/petstore/java-helidon-server/v3/mp/
           - samples/server/petstore/java-helidon-server/v3/se
-          - samples/client/petstore/spring-http-interface-reactive
-          - samples/client/petstore/spring-http-interface
```
Comment (martin-mfg) [inline .github/workflows/samples-jdk17.yaml:55]: What's the reason to remove these tests?

**Review by martin-mfg — CHANGES_REQUESTED** [review]

File: .github/workflows/samples-jdk17.yaml (hunk 2 above)
Comment (alfet81) [reply .github/workflows/samples-jdk17.yaml:55]: they are duplicated. Same paths are already present in the #clients section. Probably copy/paste issue.

File: modules/openapi-generator/src/test/resources/3_0/spring/petstore-with-fake-endpoints-models-for-testing.yaml (hunk 1 above)
Comment (alfet81) [reply modules/openapi-generator/src/test/resources/3_0/spring/petstore-with-fake-endpoints-models-for-testing.yaml:29]: Just wanted to use correct statuses for POST/PUT success responses. Gonna revert then.

**Review by martin-mfg — APPROVED** [review]

**Pull request merged**